diff --git a/symbols/de b/symbols/de
index 7eab1641..becaed44 100644
--- a/symbols/de
+++ b/symbols/de
@@ -454,9 +454,9 @@ xkb_symbols "neo_base" {
     key <AC07> { [ n,                       N,                       parenleft,               Greek_nu,                KP_4,                    KP_4,                    U2115,                   NoSymbol                 ] };
     key <AC08> { [ r,                       R,                       parenright,              Greek_rho,               KP_5,                    KP_5,                    U211D,                   NoSymbol                 ] };
     key <AC09> { [ t,                       T,                       minus,                   Greek_tau,               KP_6,                    KP_6,                    partialderivative,       NoSymbol                 ] };
-    key <AC10> { [ d,                       D,                       colon,                   Greek_delta,             KP_Separator,            comma,                   Greek_DELTA,             NoSymbol                 ] };
+    key <AC10> { [ d,                       D,                       colon,                   Greek_delta,             KP_Separator,            parenleft,               Greek_DELTA,             NoSymbol                 ] };
 
-    key <AC11> { [ y,                       Y,                       at,                      Greek_upsilon,           period,                  KP_Decimal,              nabla,                   NoSymbol                 ] };
+    key <AC11> { [ y,                       Y,                       at,                      Greek_upsilon,           KP_Decimal,              parenright,              nabla,                   NoSymbol                 ] };
 
     // Bottom row
     // --------------------------------------------------------------
